R/FilterState-utils.R

```diff
@@ -78,213 +88,294 @@ init_filter_state <- function(x,
   if (is.null(varlabel)) varlabel <- character(0L)
 
   if (all(is.na(x))) {
-    return(
-      EmptyFilterState$new(
-        x = x,
-        x_reactive = x_reactive,
-        varname = varname,
-        varlabel = varlabel,
-        dataname = dataname,
-        extract_type = extract_type
-      )
+    args <- list(
+      x = x,
+      x_reactive = x_reactive,
+      dataname = dataname,
+      varname = varname,
+      choices = choices,
+      selected = selected,
+      varlabel = varlabel,
+      keep_na = keep_na,
+      keep_inf = keep_inf,
+      fixed = fixed,
+      extract_type = extract_type
     )
+    do.call(EmptyFilterState$new, args)
+  } else {
+    UseMethod("init_filter_state")
   }
-  UseMethod("init_filter_state")
 }
 
 #' @keywords internal
 #' @export
 init_filter_state.default <- function(x,
                                       x_reactive = reactive(NULL),
-                                      varname,
-                                      varlabel = attr(x, "label"),
-                                      dataname = NULL,
+                                      dataname,
+                                      varname = attr(x, "label"),
+                                      choices = NULL,
+                                      selected = NULL,
+                                      varlabel = character(0),
+                                      keep_na = NULL,
+                                      keep_inf = NULL,
+                                      fixed = FALSE,
                                       extract_type = character(0)) {
   if (is.null(varlabel)) varlabel <- character(0)
-  InteractiveFilterState$new(
+
+  args <- list(
     x = x,
     x_reactive = x_reactive,
+    dataname = dataname,
     varname = varname,
+    choices = choices,
+    selected = selected,
     varlabel = varlabel,
-    dataname = dataname,
+    keep_na = keep_na,
+    keep_inf = keep_inf,
+    fixed = fixed,
     extract_type = extract_type
   )
+
+  do.call(InteractiveFilterState$new, args)
 }
 
 #' @keywords internal
 #' @export
 init_filter_state.logical <- function(x,
                                       x_reactive = reactive(NULL),
-                                      varname,
-                                      varlabel = attr(x, "label"),
-                                      dataname = NULL,
+                                      dataname,
+                                      varname = attr(x, "label"),
+                                      choices = NULL,
```

Comment:
What do you think?
```suggestion
                                      choices = unique(na.omit(x)),
```

Reply:
will do :+1:

Reply:
Does this go for `FilterState` constructors as well?